pkg/engine/utils.go

```diff
@@ -32,6 +32,10 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+var (
```

Comment:
This is not going to be thread safe.

Reply:
My confusion when looking at the initial code: I don't think it was thread safe before? Or am I misunderstanding the initial implementation?

Reply:
Not sure if was thread safe before, maybe not when using a `store` ?
This `store` thing being referenced in the `engine` is just crazy.

Reply:
I don't mind if it's not thread safe when used with a `store` but it should be thread safe in the general case.